### ISSUE TITLE
Generated JS on page contains unnecessary values and functions.

### ISSUE THREAD

**Issue by elordin** [links-lang/links#235]
E.g. when running the examples, the page source of an example contains the code of the other examples even though that code is in no way necessary for that page.

This bloats page size, pollutes the global scope, impacts performance, and makes debugging some JS an absolute nightmare.

**Comment by SimonJF** [comment]
Yes, that's true. It's a pain, and needs a proper design more than anything.

**Comment by elordin** [comment]
Only loosely related and more of an idea than an actual suggestion:

Modules could be translated to JS objects:
```
module Foo {
  var x = 5;
  fun bar (y) { ... }
}
```
->
```JavaScript
const Foo = {
  x: 5,
  bar: (_1234) => { something_yield_something }
};
```
